tests/test_transform.py

```diff
@@ -252,6 +253,68 @@ def test_null_object_transform(self):
         empty_data = {'addrs': {}}
         self.assertDictEqual(empty_data, transform(empty_data, schema))
 
+    def test_decimal_types_transform(self):
+        schema =  {"type": "object",
+                   "properties": {"percentage": {"type": ["string"],
+                                                 "format": "singer.decimal"}}}
+
+        inf = {'percentage': 'Infinity'}
+        negative_inf = {'percentage': '-Infinity'}
+        root2  = {'percentage': 1.4142135623730951}
+        nan = {'percentage': decimal.Decimal('NaN')}
+        snan = {'percentage': decimal.Decimal('sNaN')}
+
+        self.assertEquals(inf, transform(inf, schema))
+        self.assertEquals(negative_inf, transform(negative_inf, schema))
+        self.assertEquals({'percentage': '1.4142135623730951'}, transform(root2, schema))
+        self.assertEquals({'percentage': 'NaN'}, transform(nan, schema))
+        self.assertEquals({'percentage': 'sNaN'}, transform(snan, schema))
```

Comment:
This should come out as just `NaN`. I think the signaling designation doesn't make sense anymore once it crosses serialization boundaries to a string in JSON, and from what I'm seeing around about the difference between the two, there's no real agreed upon way of consistently representing a NaN (quiet or signaling)

This even goes so far as to talk about 2^53 -2 distinct NaN values by "bit pattern". https://stackoverflow.com/a/25051746

So, as far as seeing a NaN or sNaN in Python, converting to a regular NaN seems reasonable.